components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -445,6 +449,13 @@ private void deleteSTOREOperationsTask(Timestamp timestamp) {
             return;
         }
         try {
+            if (StringUtils.isBlank(sqlDeleteSTORETask)) {
+                if (connection.getMetaData().getDriverName().contains("MySQL")) {
+                    sqlDeleteSTORETask = SQL_DELETE_STORE_OPERATIONS_TASK_MYSQL;
+                } else {
+                    sqlDeleteSTORETask = SQL_DELETE_STORE_OPERATIONS_TASK;
+                }
+            }
```

Comment:
Can we try getting the connection and setting this in the constructor itself and then closing the connection.